### PR TITLE
Truncate (for now) too long output

### LINKED ISSUE #77: 2000 limit on replies
Ultimately we could attach the full result but for now let's just truncate

### PR DESCRIPTION
**Pull request by ldemailly**
fixes #77

### CODE REVIEW

File: bot.go [hunk 1]
```diff
@@ -269,7 +270,15 @@ func evalAndReply(session *discordgo.Session, info, channelID, input string,
 	replyID string, formatMode, compactMode, verbatimMode bool,
 ) string {
 	res := eval(input, formatMode, compactMode, verbatimMode)
-	log.S(log.Info, info, log.String("response", res))
+	level := log.Info
+	msg := "response"
+	if len(res) > 1900 {
+		// hope we didn't garbage some unicode char in the middle of a utf8 char...
```
Comment (ccoVeille) [inline bot.go:276]: You can use runes for that.

https://github.com/cdfmlr/ellipsis/blob/main/ellipsis.go#L51-L61

Another solution is to look for the last space before the 1900th character, then cut and add the ellipsis.

This doesn't work for continuous text like CJK

A common way is too look for the last space character between the 1800th and 1900th characters, if none is found you can apply your strategy. But using runes should be prefered.

Comment (ccoVeille) [reply bot.go:276]: Another solution

https://pkg.go.dev/tailscale.com/util/truncate#String

File: bot.go [hunk 2]
```diff
@@ -269,7 +270,15 @@ func evalAndReply(session *discordgo.Session, info, channelID, input string,
 	replyID string, formatMode, compactMode, verbatimMode bool,
 ) string {
 	res := eval(input, formatMode, compactMode, verbatimMode)
-	log.S(log.Info, info, log.String("response", res))
+	level := log.Info
+	msg := "response"
+	if len(res) > 1900 {
```
[the remark this answers is not in the document]
Comment (ldemailly) [reply bot.go:275]: yeah it came from an error in the logs (I updated #77) but 2000 is magic, will update though... it's 2 lines apart

File: bot.go (hunk 1 above)
Comment (ldemailly) [reply bot.go:276]: I know I can use runes :) just that I didn't feel like paying the overhead

this being said allegedly the limit is in characters/runes... so I will try and confirm if that's true

and I did test it and discord handles bad utf8 fine (they kinda have to)

Screenshot 2024-08-16 at 4 16 25 PM

File: bot.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ldemailly) [reply bot.go:275]: fixed in #79

File: bot.go (hunk 1 above)
Comment (ldemailly) [reply bot.go:276]: fixed in #79

[the remark this answers is not in the document]
Comment (ldemailly) [reply bot.go:276]: also fixed by removing that comment in #79

**Pull request merged**